tests/test_header.py

```diff
@@ -244,6 +245,21 @@ def test_find_and_replace_verbatim():
     assert find_and_replace_header(text, info) == text
 
 
+def test_find_and_replace_verbatim_no_newline():
+    """Replace a header with itself with no newline."""
```

Comment:
This docstring would benefit from being a little clearer.

```suggestion
    """Do not add an empty line after existing headers."""
```